src/Razor/src/Microsoft.AspNetCore.Razor.ExternalAccess.RoslynWorkspace/RazorWorkspaceListener.cs

```diff
@@ -38,21 +34,39 @@ private void Workspace_WorkspaceChanged(object? sender, WorkspaceChangeEventArgs
     {
         switch (e.Kind)
         {
-            case WorkspaceChangeKind.SolutionAdded:
             case WorkspaceChangeKind.SolutionChanged:
             case WorkspaceChangeKind.SolutionReloaded:
+                foreach (var project in e.OldSolution.Projects)
```

Comment:
Figured now that we're using IDs, its better to be a bit more strict with removal so we're not holding stale data in the map. Not a big deal though, as removing and re-adding the same project is essentially the same as queuing an update anyway, and any stale projects here will just be an instance of the TaskDelayScheduler that does nothing.